src/lib/nlcHelper.js

```diff
@@ -0,0 +1,346 @@
+		this.missingEnv = 'HUBOT_WATSON_NLC_USERNAME';
+	}
+	else if (!context.settings.nlc_password || context.settings.nlc_password.length === 0) {
+		this.missingEnv = 'HUBOT_WATSON_NLC_PASSWORD';
+	}
+	else if (!context.settings.nlc_objectstorage_classifier || context.settings.nlc_objectstorage_classifier.length === 0) {
+		this.missingEnv = 'HUBOT_WATSON_NLC_OJBECTSTORAGE_CLASSIFIER_NAME';
+	}
+
+	if (!this.missingEnv) {
+		this.initSuccess = true;
+	}
+
+	this.logger = context.robot.logger;
+	if (!this.logger) {
+		throw new Error('Logger required when using NLCHelper');
+	}
+
+	this.opts = {
+		url: context.settings.nlc_url,
+		username: context.settings.nlc_username,
+		password: context.settings.nlc_password,
+		version: context.settings.nlc_version,
+		classifierName: context.settings.nlc_objectstorage_classifier,
+		maxClassifiers: 1,
+		classifierLanguage: 'en'
+	};
+
+	if (!this.missingEnv)
+		this.nlc = watson.natural_language_classifier(this.opts);
+}
+
+/**
+ * Returns whether the NLCHelper initialized properly.
+ *
+ * @return Boolean true if the NLCHelper is ready to use
+ */
+NLCHelper.prototype.initializedSuccessfully = function() {
+	return this.initSuccess;
+};
+
+/**
+ * Returns the missing environment variable that caused the NLCHelper not to initialize properly
+ *
+ * @return String the missing environment variable that caused the NLCHelper not to initialize properly
+ */
+NLCHelper.prototype.getMissingEnv = function() {
+	return this.missingEnv;
+};
+
+/**
+ * Returns classification data for a statement using the latest classifier available.
+ *
+ * @param  String	text	Natural Language statement to be classified.
+ * @return JSON      		Classification data from Watson Natural Language Classifier.
+ */
+NLCHelper.prototype.classify = function(text) {
+	let dfd = Promise.defer();
+	this._getClassifier().then((classifier) => {
+		this.logger.info('Using classifier %s', JSON.stringify(classifier));
+		if (classifier.status === 'Training') {
+			dfd.reject(new Error('There is not an available classifer at this time.'));
+		}
+		else {
+			this.nlc.classify(
+				{
+					text: text,
+					classifier_id: classifier.classifier_id
+				},
+				(err, response) => {
+					if (err) {
+						this.classifier_cache = undefined;
+						dfd.reject(err);
+					}
+					else {
+						dfd.resolve(response);
+					}
+				});
+		}
+	}).catch((err) => {
+		this.classifier_cache = undefined;
+		dfd.reject(err);
+	});
+	return dfd.promise;
+};
+
+/**
+ * Helper method to finds a classifier which is available (training completed)
+ * and with the most recent creation date.  If none are 'Available' then find
+ * the most recent classifier that started training.  If none are training,
+ * start the training.
+ *
+ * @return Promise When resolved it returns a JSON object with the classifier information.
+ */
+NLCHelper.prototype._getClassifier = function() {
+	let dfd = Promise.defer();
+
+	if (this.classifier_cache) {
```

Comment:
We shouldn't be using classifier_cache.  At least not the way it's currently implemented.  The code running on the bot should be smart enough to detect new classifiers.  The cache will prevent this.